src/main/java/cubicchunks/asm/mixin/core/common/MixinWorldServer.java

```diff
@@ -72,16 +75,18 @@
 		this.isCubicWorld = true;
 
 		this.entitySpawner = new CubeWorldEntitySpawner();
-		this.chunkProvider = new ServerCubeCache(this);
-		this.generatorPipeline = new GeneratorPipeline(this);
-		this.thePlayerManager = new PlayerCubeMap(this);
-		this.chunkGc = new ChunkGc(getCubeCache(), getPlayerCubeMap());
+		ServerCubeCache chunkProvider = new ServerCubeCache(this);
+		this.chunkProvider = chunkProvider;
+		this.generatorPipeline = new GeneratorPipeline();
 		this.lightingManager = new LightingManager(this);
 
-
 		ICubicChunksWorldType type = (ICubicChunksWorldType) this.getWorldType();
 		type.registerWorldGen(this, this.generatorPipeline);
-		this.generatorPipeline.checkStages();
+		GeneratorPipeline.checkStages(this.generatorPipeline);
+		this.cubeGenerator = new WorldGenerator(this, this.generatorPipeline);
```

Comment:
According to Wikipedia:

> In software engineering, a pipeline consists of a chain of processing elements (processes, threads, coroutines, functions, etc.), arranged so that the output of each element is the input of the next.

A registry on the other hand does not have an ordering of that kind.